examples/sampleview/core/sampledesigner/SceneModelController.cpp

```diff
@@ -155,6 +214,66 @@ void SceneModelController::onModelDestroyed()
 }
 
 namespace {
+std::set<const SessionItem*> topItems(const QList<QGraphicsItem*>& views)
+{
+    std::vector<const SessionItem*> items;
+    for (auto view: views)
+        if (auto iview = dynamic_cast<IView*>(view); iview && iview->getItem())
+            items.push_back(iview->getItem());
+
+    std::set<const SessionItem*> result;
+    for (const SessionItem* item: items) {
+        auto parent = item->parent();
+        bool has_parent = false;
+        while (parent != nullptr) {
+            has_parent = std::find(items.begin(), items.end(), parent) != items.end();
+            if (has_parent)
+                break;
+            parent = parent->parent();
+        }
+        if (!has_parent)
+            result.insert(item);
+    }
+
+    return result;
+}
```

Comment:
Would be nice to refactor at some point. It is hard to understand what is going on, without deep diving into the code.